### PR TITLE
Add all components from PF4 @patternfly/react-catalog-view-extension to replace pf 3 react-extensions in console

### PR DESCRIPTION
**Pull request by jcaianirh**
Replace pf3 dependency with PF4 PropertiesSidePanel, VerticalTabs, CatalogItemHeader, CatalogTile, and FilterSidePanel to console using new pf4 extensions. All components from pf3 react extensions are now included in this pr.

VerticalTabs:
Screen Shot 2019-11-08 at 2 14 41 PM
PropertiesSidePanel:
Screen Shot 2019-11-08 at 2 16 22 PM
FilterSidePanel
Screen Shot 2019-11-08 at 2 15 35 PM

Screen Shot 2019-11-19 at 4 30 39 PM

cc: @bmignano @spadgett

### CODE REVIEW

**Comment by spadgett** [comment]
/hold
for 4.4

**Comment by spadgett** [comment]
@rhamilto fyi

File: frontend/packages/dev-console/src/components/EmptyState.tsx [hunk 1]
```diff
@@ -83,7 +83,7 @@ const ODCEmptyState: React.FC<Props> = ({
       <div className="odc-empty-state__content">
         {allImportResourceAccess && (
           <CatalogTile
-            onClick={(e: Event) => navigateTo(e, '/import?importType=git')}
+            onClick={(e: any) => navigateTo(e, '/import?importType=git')}
```
Comment (spadgett) [inline frontend/packages/dev-console/src/components/EmptyState.tsx]: Avoid `any` if possible. Does `onClick` not have types in Patternfly?

Comment (jcaianirh) [reply frontend/packages/dev-console/src/components/EmptyState.tsx]: thanks @spadgett. this pr is currently stubs out the catalog tiles. its for propertiessidepanel, verticaltabs, and filtersidepanel. @rebeccaalpert is following up with the catalog tile components.

Comment (jcaianirh) [reply frontend/packages/dev-console/src/components/EmptyState.tsx]: @spadgett All components are now added to this pr, and the above type comment has been addressed.

File: frontend/__tests__/components/catalog.spec.tsx [hunk 2]
```diff
@@ -2,9 +2,9 @@ import * as React from 'react';
 import * as _ from 'lodash-es';
 import { mount, ReactWrapper } from 'enzyme';
 
-import { CatalogTile } from '../../node_modules/patternfly-react-extensions/dist/js/components/CatalogTile';
-import { VerticalTabsTab } from '../../node_modules/patternfly-react-extensions/dist/js/components/VerticalTabs';
-import { FilterSidePanel } from '../../node_modules/patternfly-react-extensions/dist/js/components/FilterSidePanel';
+import { CatalogTile } from '@patternfly/react-catalog-view-extension/dist/js/components/CatalogTile';
```
Comment (redallen) [inline frontend/__tests__/components/catalog.spec.tsx]: Do you have to do the ugly `/dist/js/components/CatalogTile` or are we not properly exporting `CatalogTile`, `VerticalTabsTab`, and `FilterSidePanelCategoryItem` upstream?

Comment (rebeccaalpert) [reply frontend/__tests__/components/catalog.spec.tsx]: This has been fixed.

Comment (redallen) [reply frontend/__tests__/components/catalog.spec.tsx]: 🏅

**Comment by rebeccaalpert** [comment]
```
displays subscription creation form for selected Operator expand_less | 3m0s
-- | --
Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.     at Timeout._onTimeout (/go/src/github.com/openshift/console/frontend/node_modules/jasmine/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:4281:23)     at ontimeout (timers.js:427:11)     at tryOnTimeout (timers.js:289:5)     at listOnTimeout (timers.js:252:5)     at Timer.processTimers (timers.js:212:10)
```

/retest

**Comment by rebeccaalpert** [comment]
Same error. Investigating.

**Comment by spadgett** [comment]
> Same error. Investigating.

It's a known flake :(

File: frontend/public/components/utils/tile-view-page.jsx [hunk 3]
```diff
@@ -1,14 +1,22 @@
...
 import * as PropTypes from 'prop-types';
-import { FilterSidePanel, VerticalTabs } from 'patternfly-react-extensions';
+import {
+  FilterSidePanel,
+  FilterSidePanelCategory,
+  FilterSidePanelCategoryItem,
+  VerticalTabs,
+  VerticalTabsTab,
+} from '@patternfly/react-catalog-view-extension';
 import { FormControl } from 'patternfly-react';
```
Comment (rhamilto) [inline frontend/public/components/utils/tile-view-page.jsx]: While you're in here, mind migrating`<FormControl>` to a plain HTML element so we can eliminate this patternfly-react dependency?

Comment (rebeccaalpert) [reply frontend/public/components/utils/tile-view-page.jsx]: I can give this a go. @jeff-phillips-18 raised an issue about how the catalog tiles may be better off with more programmatic description heights given how much the content can vary, so there may be some more changes here anyway as I work on that.

Comment (rhamilto) [reply frontend/public/components/utils/tile-view-page.jsx]: Thanks, @rebeccaalpert.  It should be as simple as:
```javascript
              <input
                className="pf-c-form-control co-catalog-page__input"
                type="text"
                ref={(ref) => (this.filterByKeywordInput = ref)}
                placeholder="Filter by keyword..."
                value={activeFilters.keyword.value}
                onChange={(e) => this.onKeywordChange(e.target.value)}
```

Comment (rebeccaalpert) [reply frontend/public/components/utils/tile-view-page.jsx]: I have a PR up in Joe's branch for the input; should be in soonish.

**Comment by jcaianirh** [comment]
@spadgett could you approve this pr please?

File: frontend/packages/operator-lifecycle-manager/src/style.scss [hunk 4]
```diff
@@ -275,3 +275,7 @@ $co-affinity-row-margin: 15px;
 .key-operator-value__row {
   padding-bottom: 15px;
 }
+
+.co-catalog-tile {
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/style.scss]: nit: alpha

**Pull request merged**